### ISSUE TITLE
No error on bad credentials

### ISSUE THREAD

**Issue by victornikitin** [vesse/passport-ldapauth#87]
How to catch that bindDN or bindCredentials are incorrect?

I pass authenticate to express via router:
```
router.use(
    passport.authenticate('ldapauth'),
    function(err, req, res, next) {
      // this code is not called when server credentials are wrong
      // only 403 forbidden is returned to user
      console.log(err);
      next();
    }
  );
```

Even if I create strategy with `handleErrorsAsFailures` option, `failureErrorCallback` does not trigger too.

How to catch this situation?
Or what I'm doing wrong?

**Comment by vesse** [comment]
You are apparently trying to use [custom callbacks](http://www.passportjs.org/docs/authenticate/#custom-callback), see that on how the middleware needs to be called then.

**Comment by victornikitin** [comment]
You are right. I've changed code to this:
```
passport.use("ldapauth", LDAPStrategy({...}, authenticateLdap) );
function authenticateLdap(profile, done) {
    if (!profile) {
      done(null, false, { message: options.incorrectPasswordError });
      return;
    }
    ...
}
...
router.use(function(req, res, next) {
    passport.authenticate('ldapauth', async function(err, user, message) {
      if (!user) {
       // user password is wrong or binddn credentials is wrong?
      }
      req.logIn(user, function(err) {
       next();
      });
    })(req, res, next);
  });
```

But how to check if it user password is incorrect or bindDN credentials are incorrect?

**Comment by vesse** [comment]
The strategy does not provide such info as this is more of a configuration error. It is sort of expected that admin credentials are provided correctly. You can pass a Bunyan logger instance to the underlying ldapauth-fork library and see the trace for bind errors, or try eg. with command line tool ldapsearch to see if your admin credentials are correct.

**Comment by clarkey** [comment]
@victornikitin Hi. I came across this issue just now. Did you come up with anything?

**Comment by victornikitin** [comment]
No, we did not found any good solutions how to determine is this a user
credentials error or a connection error.

**Comment by MarkusRissmann** [comment]
I'm having the same issue. No matter what I am getting "Unauthorized". I wrote a quick c# .net app using all the same settings and it works. I know my code works because I've used it against another ldap server. Kind of hard to ask IT to fix something when you don't know what is wrong.
